pip/pep425tags.py

```diff
@@ -198,7 +247,8 @@ def get_supported(versions=None, noarch=False):
                 supported.append(('%s%s' % (impl, versions[0]), abi, arch))
 
         # Has binaries, does not use the Python API:
-        supported.append(('py%s' % (versions[0][0]), 'none', arch))
+        for arch in arches:
+            supported.append(('py%s' % (versions[0][0]), 'none', arch))
```

Comment:
This is a partial fix to an issue described in #3202; I have a PR open with a full fix at #3357.